internal/api/client.go

```diff
@@ -380,3 +380,132 @@ func (c *Client) SetClusterInstallationConfig(clusterInstallationID string, conf
 		return errors.Errorf("failed with status code %d", resp.StatusCode)
 	}
 }
+
+// CreateGroup requests the creation of a group from the configured provisioning server.
+func (c *Client) CreateGroup(request *CreateGroupRequest) (*model.Group, error) {
+	resp, err := c.doPost(c.buildURL("/api/groups"), request)
+	if err != nil {
+		return nil, err
+	}
+	defer closeBody(resp)
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		return model.GroupFromReader(resp.Body)
+
+	default:
+		return nil, errors.Errorf("failed with status code %d", resp.StatusCode)
+	}
+}
+
+// UpdateGroup updates the installation group.
+func (c *Client) UpdateGroup(request *PatchGroupRequest) error {
+	resp, err := c.doPut(c.buildURL("/api/group/%s", request.ID), request)
+	if err != nil {
+		return err
+	}
+	defer closeBody(resp)
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		return nil
+
+	default:
+		return errors.Errorf("failed with status code %d", resp.StatusCode)
```

Comment:
super nit: this error message is used heavily throughout the code. Perhaps we may want a helper function to return it in a standard way to make updating it in the future a little easier. 0/5 on this.

Reply:
I agree that this needs some attention. I'd like to also expose the actual error message to the client vs. requiring the server logs to explain. I'll add this as a TODO to revisit holistically.